### PR TITLE
Feat : add Homepage and single product

### PR DESCRIPTION
**Pull request by ZuberDunge**

### CODE REVIEW

**Review by Prabhav-Sharma — APPROVED** [review]
Good practices implemented!

**Review by Hamdrive — COMMENTED** [review]
Great work done, liked the implementation of context and reducer
Will suggest to try to use async await for Axios calls

File: src/frontend/style/nav.css [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+    .nav-menu button:hover {
+        background-color: transparent;
+    }
+
+    .hide-on-phone {
+        display: none;
+    }
+
+    .nav-menu {
+        /* position: fixed; */
```
Comment (nerdyspook) [inline src/frontend/style/nav.css:121]: You can remove the commented code here.

**Review by nerdyspook — APPROVED** [review]
Structure of the code is really good.

**Pull request merged**